Src/GoogleApis.Auth.DotNet4/OAuth2/DefaultCredentialProvider.cs

```diff
@@ -0,0 +1,255 @@
+                    return new ComputeCredential();
+                }
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Converts a stream to a credential object.
+        /// </summary>
+        internal static ICredential LoadFromStream(Stream stream)
+        {
+            ClientCredentialParameters clientCredentialParameters = ClientCredentialParameters.Load(stream);
+          
+            switch (clientCredentialParameters.CredentialType)
+            {
+                case ClientCredentialType.UserCredential:
+                    try
+                    {
+                        var initializer = new GoogleAuthorizationCodeFlow.Initializer
+                        {
+                            ClientSecrets = clientCredentialParameters.ClientSecrets
+                        };
+
+                        TokenResponse token = new TokenResponse()
+                        {
+                            RefreshToken = clientCredentialParameters.RefreshToken
+                        };
+
+                        var flow = new GoogleAuthorizationCodeFlow(initializer);
+                        return new UserCredential(flow, "ApplicationDefaultCredentials", token);
+                    }
+                    catch (IOException e)
+                    {
+                        throw new InvalidOperationException("Error parsing stream contents.", e);
+                    }
+
+                case ClientCredentialType.ServiceAccountCredential:
+                    try
+                    {
+                        var initializer = new ServiceAccountCredential.Initializer(clientCredentialParameters);
+                        return new ServiceAccountCredential(initializer);
+                    }
+                    catch (IOException e)
+                    {
+                        throw new InvalidOperationException("Error parsing stream contents.", e);
+                    }
+
+                default:
+                    throw new InvalidOperationException("Error reading credentials from stream, 'type' field not specified.");
+            }
+        }
+
+        /// <summary> 
+        /// Helper method to return well known credential file path. This file is created by 'gcloud auth login'.
+        /// </summary>
+        private string GetWellKnownCredentialFilePath()
+        {
+            return Path.Combine(System.Environment.GetEnvironmentVariable(APPDATA_ENV_VAR),
+                CLOUDSDK_CONFIG_DIRECTORY,
+                WELL_KNOWN_CREDENTIALS_FILE);
+        }
+
+        /// <summary>
+        /// Helper method to load file stream.
+        /// </summary>
+        internal ICredential LoadCredentialFromFile(string credentialPath)
+        {
+            Logger.Info(String.Format("Loading Credential from file {0}", credentialPath));
+
+            using (Stream fs = GetStream(credentialPath))
```

Comment:
why fs? I would just rename it to stream.
You can even use the following:
using (var stream = GetStream(credetialPath))
...